include/PathUtilities.h

```diff
@@ -0,0 +1,289 @@
+    Boole MEZZ_LIB IsPathRelative_Posix(const StringView ToCheck);
+    /// @brief Gets whether or not a path is relative on Windows.
+    /// @warning This method is imperfect and doesn't check for some faults that can occur in a path, nor does
+    /// it check for proper Windows style directory separators as most uses of filesystem paths in the Windows
+    /// API can tolerate Windows or Posix syle.
+    /// @note A path is relative if it requires using the current working directory to define the location of
+    /// a resource.
+    /// @param ToCheck The path to check.
+    /// @return Returns true if the specified path is considered relative on a Windows platform, false otherwise.
+    Boole MEZZ_LIB IsPathRelative_Windows(const StringView ToCheck);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Path Checks
+
+    /// @brief Gets how many directories deep a path is.
+    /// @param ToCheck The path to get the depth of.
+    /// @param ExitIfNegative If true, the function to return immediately if the depth count becomes negative.
+    /// @return Returns an Integer representing how many directories down (or up, if negative) the path goes.
+    Integer MEZZ_LIB GetDirectoryDepth(const StringView ToCheck, const Boole ExitIfNegative);
+    /// @brief Checks whether or not a given path is a subdirectory or file of a base path.
+    /// @note This method will make no attempt to remove dot segments or perform any other normalization
+    /// routine to the provided path, thus these transformations must be done manually before calling this
+    /// method if they are needed.
+    /// @exception This function will only operate if both paths are absolute, or both are relative.  Mixing
+    /// of absolute and relative is not supported by this function.  Attempting to compare an absolute and
+    /// relative path will result in an PARAMETERS_EXCEPTION being thrown.
+    /// @param BasePath The base path to check.
+    /// @param CheckPath The path being checked against base path to see if it points to a subdirectory or file.
+    /// @return Returns true if CheckPath is a subdirectory or file of BasePath, false otherwise.
+    Boole MEZZ_LIB IsSubPath(const StringView BasePath, const StringView CheckPath);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Path Utilities
+
+    /// @brief Builds a String path based on a number of directory/file names in a String vector.
+    /// @note This function relies on the PathRoot argument to generate absolute paths.
+    /// @param PathRoot This string will be prepended to the result of the ToBuild parameter.
+    /// @param ToBuild A vector of Strings containing the overall path to be built.
+    /// @param FileName The file portion of the path to be built.
+    /// @param UseWindowsSlash If true, path will be generated with '\\' separators. Otherwise '/' will be used.
```

Comment:
Should UseWindowsSlash default to false since the posix slash works most places and usually works on windows?

Reply:
Defaulted.